Updated docs for changes to progress calculation (% complete)

psatyal opened the pull request:
Docs WIP

- [x] Write draft for new progress calculation feature
- [x] Update: Work pacakge status page
- [x] Update: Work package settings page
- [x] Add details re: scenarios when switching between status and work modes
- [x] Get it reviewed by Christophe
- [x] Add screenshots
- [ ] Final review by Marketing

https://community.openproject.org/work_packages/53755/activity

psatyal committed:
First draft of updated docs page

Still need to add edge cases

psatyal committed:
Updates

psatyal committed:
Updating status and work package settings page

psatyal committed:
Added details on mode switch scenarios

psatyal committed:
Updated based on @cbliard's suggestions

cbliard commented on docs/system-admin-guide/manage-work-packages/work-package-settings/README.md:
```diff
@@ -17,7 +17,9 @@ You can adjust the following:
 3. **Use current date as start date for new work packages**. This way the current date will always be set as a start date if your create new work packages. Also, if you copy projects, the new work packages will get the current date as start date.
 
-4. **Calculate the work package done ratio with** ... defines how the **Progress %** field is calculated for work packages. If you choose "disable", the field will not be shown. If you select "Use the work package field", the Progress % field can be manually set in 10% steps directly in the work package attribute. If you opt for „Use the work package status", the Progress % field is chosen based  ...
+4. **Progress calculation** let you pick between two modes for how the **%&nsbp;Complete** field is calculated for work packages.
```
```suggestion
4. **Progress calculation** lets you pick between two modes for how the **%&nsbp;Complete** field is calculated for work packages.
```

cbliard commented on docs/system-admin-guide/manage-work-packages/work-package-settings/README.md:
```diff
@@ -17,7 +17,9 @@ You can adjust the following:
 
-4. **Calculate the work package done ratio with** ... defines how the **Progress %** field is calculated for work packages. If you choose "disable", the field will not be shown. If you select "Use the work package field", the Progress % field can be manually set in 10% steps directly in the work package attribute. If you opt for „Use the work package status", the Progress % field is chosen based  ...
+4. **Progress calculation** let you pick between two modes for how the **%&nsbp;Complete** field is calculated for work packages.
+	- **Work-based**: %&nbsp;Complete is automatically calculated based on Work and Remaining work values for that work package (both of which are then necessary to have a value for %&nbsp;Complete).
```
```suggestion
	- **Work-based**: %&nbsp;Complete is automatically calculated based on Work and Remaining work values for that work package, both of which are then necessary to have a value for %&nbsp;Complete.
```

cbliard commented on docs/system-admin-guide/manage-work-packages/work-package-status/README.md:
```diff
@@ -19,16 +19,17 @@ To learn about the options for configuring the transition from one status to ano
 
 To create a new work package status click the green ***+ New Status*** icon. 
 
-![work package status](image-20200211142716298.png)
```
The image `docs/system-admin-guide/manage-work-packages/work-package-status/image-20200211142716298.png` should be deleted as part of this PR.

cbliard commented on docs/system-admin-guide/manage-work-packages/work-package-status/README.md:
```diff
@@ -19,16 +19,17 @@ To learn about the options for configuring the transition from one status to ano
 
 To create a new work package status click the green ***+ New Status*** icon. 
```
Incorrect button label: it's "+ Status", not "+ New Status".

psatyal replied:
Fixed.

cbliard commented on docs/system-admin-guide/manage-work-packages/work-package-status/README.md:
```diff
@@ -19,16 +19,17 @@ To learn about the options for configuring the transition from one status to ano
+
+Save the new status by clicking **Create**.
 
 ![Sys-admin-work-package-new-status](Sys-admin-work-package-new-status.png)
```
Screenshot is not up-to-date yet. Is it expected?

MayaBerd replied:
@cbliard it is indeed outdated. I've updated it.

cbliard commented on docs/system-admin-guide/manage-work-packages/work-package-status/README.md:
```diff
@@ -19,16 +19,17 @@ To learn about the options for configuring the transition from one status to ano
+4. Define if this status is set as default value when creating new work packages. BEWARE: If you decide to set the new status as default it will impact all work packages, existing and future ones. See more below.
+5. Check if this status sets a **work package in read-only mode**. This means no work package attributes can be changed except the status.
+6. Set a **color** for the work package status. The color appears in the status button of the work packages or in the work package table.
+7. Check **Exclude from calculation of totals in hierarchy** if you do not want work packages with this status to *not* be included in the calculation of sums in a hierarchy. This is useful for statuses like *closed* or *rejected*.
```
This should be removed. "Exclude from calculation of totals in hierarchy" is not availabl for 14.0.

psatyal replied:
Commented out.

cbliard commented on docs/system-admin-guide/manage-work-packages/work-package-status/README.md:
```diff
@@ -19,16 +19,17 @@ To learn about the options for configuring the transition from one status to ano
-
 A new window will open, where you will be able to specify the following:
 
 1. Name the new work package status.
```
As a side note, is it normal that the order of these fields in create status page is different from the order of the columns in the statuses index page?

MayaBerd replied:
Excellent question. My first response is yes, but I will follow-up to make sure and will update the screenshots if needed.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
 
-Clicking on the progress bar opens an in-place editor
-that allows you to enter a percentage value between 0 and 100.
+%&nbsp;Complete is an automatically calculated value that is a function of Work and Remaining work (unless %&nbsp;Complete is configured to be [set by status](#status-based-progress-reporting).
```
```suggestion
%&nbsp;Complete is an automatically calculated value that is a function of Work and Remaining work (unless %&nbsp;Complete is configured to be [set by status](#status-based-progress-reporting)).
```
Missing closing paren. 
Could also be rewritten without parens and just a comma.

psatyal replied:
Fixed, with a comma.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
+This allows you to edit Work or Remaining work and get a preview of the updated %&nbsp;Complete value before saving changes. Changing any one field will automatically update the other two.
 
-![Bulk editing progress](progress-tracking-bulk-editing.png)
+>**Note:** If you enter a value for Remaining work that is higher than Work, you will see an error message telling you that this is not possible. You will have to enter a value lower than work to be able to save the new value.
```
```suggestion
>**Note:** If you enter a value for Remaining work that is higher than Work, you will see an error message telling you that this is not possible. You will have to enter a value lower than Work to be able to save the new value.
```

psatyal replied:
Fixed manually, thanks.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
-As an alternative to the manual progress tracking mode above, you can configure your OpenProject system to associate work packages statuses with fixed values for % Complete .
-In the Administration, please navigate to Work packages -> Settings and change “Calculate the work package done ratio” to
-“Use the work package status”.
+In Status-based progress reporting mode, Work is not a required value. However, if Work is set, Remaining work is automatically calculated. To make this link clear and transparent, clicking on any of the three values to modify them will display the following pop-over:
```
Actually, clicking on Remaining work or % Complete field will do nothing because they are read-only. Only clicking on Work will display the popover.

psatyal replied:
Fixed manually.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
-**% Complete (earlier called Progress (%))** field in the work packages.
-Instead, the **% Complete** value will be set automatically based on the work package
-status configuration page.
+This allows you to edit %&nbsp;Complete (by changing status) or Work and get a preview of the updated Remaining work before saving changes.
```
```suggestion
This allows you to edit either %&nbsp;Complete by changing status, or Work, and get a preview of the updated Remaining work before saving changes.
```

I don't like parens :)

Regarding [preview](https://community.openproject.org/projects/openproject/work_packages/54275/activity), so far it's not part of 14.0 yet.

psatyal replied:
Ignoring this particular suggestion since it doesn't read as easily (paranthesis help here), but appreciate the suggestion :) Any change it will be? (I'll leave it for now)

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
+This allows you to edit %&nbsp;Complete (by changing status) or Work and get a preview of the updated Remaining work before saving changes.
 
-Within the Administration, navigate to Work package -> Status and select the status you want to edit regarding the progress tracking. Here you can select the desired progress percentage from the drop-down menu in the progress line and add it to the status. Please do not forget to save your changes.
+>**Note:** In the upcoming version, statuses cannot have an empty %&nbsp;Complete values in status-based progress reporting mode.When upgrading, all statuses that do not have a value will take the default value of 0%.
```
```suggestion
>**Note:** In the upcoming version, statuses cannot have an empty %&nbsp;Complete value in status-based progress reporting mode. When upgrading, all statuses that do not have a value will take the default value of 0%.
```

psatyal replied:
Fixed manually, thanks.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
+
+## Hierarchy totals
+
+OpenProject will automatically show totals for Work, Remaining work and % Complete in a work package hierarchy (any parent with children). These appears in a work package table as a number with a Σ sign next to it, indicating that it is a sum of the values of the parent _and_ children.
```
```suggestion
OpenProject will automatically show totals for Work, Remaining work and % Complete in a work package hierarchy (any parent with children). These appear in a work package table as a number with a Σ sign next to it, indicating that it is a sum of the values of the parent _and_ children.
```

psatyal replied:
Fixed manually, thanks.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
+
+**[IMG: Hierarchy example]**
+
+> **Note**: The total %&nbsp;Complete value of a hierarchy is a weighted average tied to Work. For example, a feature with Work set to 50h that is 30% done will advance the sum of %&nbsp;Complete of the parent more than a feature with Work set to 5h that is 70% done. 
```
```suggestion
> **Note**: The total %&nbsp;Complete value of a hierarchy is a weighted average tied to Work. For example, a feature with Work set to 50h that is 30% done will influence the sum of %&nbsp;Complete of the parent more than a feature with Work set to 5h that is 70% done. 
```

psatyal replied:
Fixed, thanks.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
+> **Note**: The total %&nbsp;Complete value of a hierarchy is a weighted average tied to Work. For example, a feature with Work set to 50h that is 30% done will advance the sum of %&nbsp;Complete of the parent more than a feature with Work set to 5h that is 70% done. 
+
+
+### Excluding certain work packages from totals
+
+In some cases, you might want to exclude certain work packages (like those with status *rejected*) from sum calculations of the parent. Administrators can define these exclusions by going to the [Administration settings for any status] and check a new option called "Exclude from calculation of totals in hierarchy". All work packages with this status will then be excluded when calculating the sum  ...
+
+A small info icon will appear next to excluded values to remind you of this fact:
+
+**[IMG: Excluded work packages warning]**
```
Excluding statuses is not available in 14.0

cbliard replied:
And we should talk about "total calculations" rather than "sum calculations" as total is the term we use in our app.

psatyal replied:
Replaced the words "sum" with "total", thanks.

psatyal replied:
And commented out the excluded statuses bit for now.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
+
+## Changing modes 
+
+When an administrator changes modes from Status-based or Work-based or vice-versa, the % Complete values might be in a "transitional stage". 
```
Actually, only when changing from work based to status based:
- when changing from work-based to status-based, % complete is applied and remaining work is derived;
- when changing from status-based to work-based, nothing changes as the values are already consistent, so there is no "transitional stage".

psatyal replied:
Rephrased and reorganised, thanks.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
+
+In Status-based mode, it is possible for work packages to have a %&nbsp;Complete value (defined by the status) without having values for Work or Remaining work. In other words, the Work and Remaining work can be empty. 
+
+However, when switching to work-based mode, work packages with a %&nbsp;Complete value also need to have values for Work and Remaining work (since the %&nbsp;Complete is not directly editable but is rather derived from the other two values). 
+
+In this case, OpenProject will retain the original value for %&nbsp;Complete that was there mode change, but will not allow you to edit it. Instead, you will be able to input Work and Remaining work, thereby overwriting the previous value with a new computed value:
+
```
```suggestion
When switching to Work-based mode, OpenProject will retain the value for %&nbsp;Complete that was set with status. It can then be modified by inputing Work and Remaining work, thereby overwriting the previous value with a new computed value:
```

I found the initial text too complicated to read.

If you do not retain this text, then there are some missing words in "that was there mode change". It does not sound correct to me.

psatyal replied:
Retained! Thanks.

cbliard reviewed (commented):
Here are some fixes and suggestions.

Everything you wrote is correct 👍 Just that having statuses excluding work packages for totals calculations is not available yet in 14.0.

cbliard commented on docs/user-guide/time-and-costs/progress-tracking/README.md:
```diff
@@ -2,93 +2,111 @@
 
-Please note the **Work** column to the right.
-Here, values next to the sum icon (for example 43 h next to the  Main task) indicate the sum of estimated time of the task itself and its children, while values without the sum icon (3 hours) refer to estimated time directly assigned to a work package.
+In this mode, each status is associated with a fixed %&nbsp;Complete value in the [admin settings for each one]((../../../system-admin-guide/manage-work-packages/work-package-status/)), which can be freely updated by changing the status of a work package. This allows teams to report progress simply by changing the status of their work packages over time.
```
```suggestion
In this mode, each status is associated with a fixed %&nbsp;Complete value in the [admin settings for each one](../../../system-admin-guide/manage-work-packages/work-package-status/), which can be freely updated by changing the status of a work package. This allows teams to report progress simply by changing the status of their work packages over time.
```

Removed extra ()

cbliard committed:
Fix broken link

MayaBerd committed:
Update docs/system-admin-guide/manage-work-packages/work-package-settings/README.md

Co-authored-by: Christophe Bliard <c.bliard@openproject.com>

MayaBerd committed:
Update docs/system-admin-guide/manage-work-packages/work-package-settings/README.md

Co-authored-by: Christophe Bliard <c.bliard@openproject.com>

psatyal committed:
Added screenshots

The warning for excluded statuses is a work package; I don't think that's been merged yet on stage.

psatyal committed:
Merge branch 'progress-14.0-doc-update' of https://github.com/opf/openproject into progress-14.0-doc-update

psatyal committed:
Uncommented the bit about excluding from totals

psatyal committed:
Fixes based on @cbliard's suggestions.

MayaBerd committed:
Update %complete docs screenshots

MayaBerd committed:
Merge branch 'release/14.0' into progress-14.0-doc-update

MayaBerd reviewed (approved):
Exchanged some outdated screenshots
